src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -182,10 +187,148 @@ public void delete(Long examId) {
+            }
+
+            List<String> containedExercises = new ArrayList<>();
+
+            for (Exercise exercise : exerciseGroup.getExercises()) {
+                containedExercises.add(exercise.getTitle().trim());
+            }
+
+            scores.exerciseGroups.add(new ExamScoresDTO.ExerciseGroup(exerciseGroup.getId(), exerciseGroup.getTitle(), maximumNumberOfPoints, containedExercises));
+        }
+
+        // Adding registered student information to DTO
+        for (User user : exam.getRegisteredUsers()) {
+            scores.studentResults.add(new ExamScoresDTO.StudentResult(user.getId(), user.getName(), user.getEmail(), user.getLogin(),
+                    user.getRegistrationNumber() != null ? user.getRegistrationNumber().trim() : ""));
+        }
+
+        List<StudentParticipation> studentParticipations = exam.getExerciseGroups().stream().map(ExerciseGroup::getExercises).flatMap(Collection::stream)
+                .map(Exercise::getStudentParticipations).flatMap(Collection::stream).collect(Collectors.toList());
+
+        // Adding student results information to DTO
+        for (ExamScoresDTO.StudentResult studentResult : scores.studentResults) {
+            // ToDo Support Team Exercises
+            List<StudentParticipation> participationsOfStudent = studentParticipations.stream()
+                    .filter(studentParticipation -> studentParticipation.getStudent().get().getId() == studentResult.id).collect(Collectors.toList());
+
+            for (StudentParticipation studentParticipation : participationsOfStudent) {
+                Exercise exercise = studentParticipation.getExercise();
+
+                Optional<Result> relevantResult = getRelevantResult(studentParticipation);
+
+                if (relevantResult.isPresent()) {
+                    Result result = relevantResult.get();
+                    Double achievedPoints = round((result.getScore() / 100.0 * exercise.getMaxScore()), 2);
+
+                    if (studentResult.overallPointsAchieved == null) {
+                        studentResult.overallPointsAchieved = achievedPoints;
+                    }
+                    else {
+                        studentResult.overallPointsAchieved = studentResult.overallPointsAchieved + achievedPoints;
+                    }
+
+                    studentResult.exerciseGroupIdToExerciseResult.put(exercise.getExerciseGroup().getId(),
+                            new ExamScoresDTO.ExerciseResult(exercise.getId(), exercise.getTitle(), exercise.getMaxScore(), result.getScore(), achievedPoints));
+                }
+            }
+        }
+
+        // Updating student result information in DTO
+        for (ExamScoresDTO.StudentResult studentResult : scores.studentResults) {
+            if (studentResult.overallPointsAchieved != null) {
+                studentResult.overallScoreAchieved = round((studentResult.overallPointsAchieved / scores.maxPoints) * 100.0, 2);
```

Comment:
scores.maxPoints is equalTo exam.maxPoints. The assumption here is, that the exam.maxPoints = sum(all exercise.maxPoints) but at the moment the scores of the exercises are set independently.

Reply:
I think we should take exam.maxPoints. The instructor needs to make sure that this is set correctly